component_plugin_template_my_account/src/my-account-components/CustomMapComponent.tsx

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable react-refresh/only-export-components */
+import React from 'react';
+import { BaseBizConsoleCompProps, SubSettingsErrorsLogger, isValidObjectValue, isValidNumberValue } from '@moderepo/bizstack-console-sdk';
+import { Box, SxProps, Theme, useTheme } from '@mui/material';
+import { Map as GoogleMap } from '@vis.gl/react-google-maps';
+
+export interface CustomMapComponentProps extends BaseBizConsoleCompProps {
+    readonly sx?: SxProps<Theme>;
+    readonly latitude: number;
+    readonly longitude: number;
+    readonly zoom: number;
+}
+
+/**
+ * Type guard function to check if the given obj is an instance of CustomMapComponentProps.
+ */
+export const isCustomMapComponentProps = (obj: unknown, errorLogger?: SubSettingsErrorsLogger | undefined): obj is CustomMapComponentProps => {
+    const settings = obj as CustomMapComponentProps;
+
+    if (!isValidObjectValue(settings, true, errorLogger)) {
+        return false;
+    }
+
+    if (
+        !isValidNumberValue(
+            settings.latitude,
+            true,
+            false,
+            -180,
+            180,
+            undefined,
+            new SubSettingsErrorsLogger(errorLogger, 'latitude', settings.latitude)
+        )
+    ) {
+        return false;
+    }
+
+    if (
+        !isValidNumberValue(
+            settings.longitude,
+            true,
+            false,
+            -180,
+            180,
+            undefined,
+            new SubSettingsErrorsLogger(errorLogger, 'longitude', settings.longitude)
+        )
+    ) {
+        return false;
+    }
+
+    if (!isValidNumberValue(settings.zoom, true, false, 0, 22, undefined, new SubSettingsErrorsLogger(errorLogger, 'zoom', settings.zoom))) {
+        return false;
+    }
+    return true;
+};
+
+export const CustomMapComponent: React.FC<CustomMapComponentProps> = ({ sx, latitude, longitude, zoom }) => {
+    // TODO FIXME useTheme doesn't recognize the BizStack Console theme that extends the MUI theme via module argumentation
+    const theme = useTheme();
+    // eslint-disable-next-line
+    const mapId = (theme as any).mapId;
```

Comment:
I couldn't find a suitable solution for this issue. We need to fix this issue later.